src/GitVersionCore/GitVersionCacheKeyFactory.cs

```diff
@@ -23,10 +25,104 @@ private static string GetGitSystemHash(GitPreparer gitPreparer, IFileSystem file
         {
             var dotGitDirectory = gitPreparer.GetDotGitDirectory();
 
-            // Maybe using timestamp in .git/refs directory is enough?
-            var lastGitRefsChangedTicks = fileSystem.GetLastDirectoryWrite(Path.Combine(dotGitDirectory, "refs"));
+            // traverse the directory and get a list of files, use that for GetHash
+            var traverse = TraverseTree(Path.Combine(dotGitDirectory, "refs"));
 
-            return GetHash(dotGitDirectory, lastGitRefsChangedTicks.ToString());
+            return GetHash(traverse.ToArray());
+        }
+
+        // lifted from https://msdn.microsoft.com/en-us/library/bb513869.aspx
+        public static List<string> TraverseTree(string root)
```

Comment:
No

> On Jul 24, 2016, at 11:07, Asbjørn Ulsberg <notifications@github.com> wrote:
> 
> In src/GitVersionCore/GitVersionCacheKeyFactory.cs:
> 
> >  
> > -            return GetHash(dotGitDirectory, lastGitRefsChangedTicks.ToString());
> > +            return GetHash(traverse.ToArray());
> > +        }
> > +
> > +        // lifted from https://msdn.microsoft.com/en-us/library/bb513869.aspx
> > +        public static List<string> TraverseTree(string root)
> Does this need to be public?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>